content/docs/reference/buildpack-api.md

```diff
@@ -187,13 +186,120 @@ The schema is as follows:
+* During the `detect` phase, the buildpack(s) may write something it `requires` or `provides` (or both) into the Build Plan.
+* During the `build` phase, the buildpack(s) may read the Buildpack Plan (a condensed version of the Build Plan, composed by the lifecycle) to determine what it should do, and refine the Buildpack Plan with more exact metadata (eg: what version dependency it requires).
+
+A buildpack can `require` or `provide` multiple dependencies, and even multiple groupings of dependencies (using `or` lists). Additionally, multiple buildpacks may `require` or `provide` the same dependency.
+
+The lifecycle uses the `Build Plan` as one element in deciding whether or not a particular list of buildpacks is appropriate, by seeing whether all dependencies required can be provided by that list. See the [spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#phase-1-detection) for particulars on how ordering buildpacks can adjust detection results.
+
+### Example
+Let's walk through some possible cases a `node-engine` buildpack may consider:
+
+1.  Nothing in the app explicitly calls out that it is needed
+2.  It is explicitly referred to in some configuration file
+3.  Two different configuration files request different versions of `node`
+
+We will also consider what a `NPM` buildpack may do.
+
+#### 1. No Explicit Request
+A `node-engine` buildpack is always happy to `provide` the `node` dependency. The build plan it will write may look something like:
+```
+[[provides]]
+name = "node"
+```
+
+#### 2. One Version Requested
+The buildpack sees in one configuration file (e.g. `.nvmrc`) that it is explicitly requested by the application. Seeing that, it may write the below text to the build plan:
+```
+[[provides]]
+name = "node"
+
+[[requires]]
+name = "node"
+version = "10.x"
+
+[requires.metadata]
+version-source = ".nvmrc"
+```
+
+As always, the buildpack `provides` `node`. In this particular case, a version of `node` (`10.x`) is being `requested` in a configuration file (`.nvmrc`), and the buildpack chooses to add an additional piece of metadata (`version-source`), so that it can understand where that request came from.
+
+#### 3. Two Versions Requested
+The buildpack sees two configuration files (e.g. `.nvmrc` and `buildpack.yml`) that request two different versions of the `node` dependency. One way of dealing with that, would be writing both `requests` to the build plan, and reconciling which version(s) it will provide during the `build` phase. It can provide two different `requires`:
+
+```
+[[provides]]
+name = "node"
+
+[[requires]]
+name = "node"
+version = "10.x"
+
+[requires.metadata]
+version-source = ".nvmrc"
+
+[[requires]]
```

Comment:
I expected this to need an `[or]` otherwise it would make a statement that it requires a node version or `10.x` AND `12.x`. Is that not the case? (Not saying it's inaccurate. Honestly, I don't know the right answer.)

Reply:
That's what I would have thought, but not necessarily. The [Paketo Node-Engine CNB](https://github.com/paketo-buildpacks/node-engine/blob/master/node/detect.go) does it this way. I find that clearer than with an `[or]`

Reply:
@sclevine Would there be a preference between using two different requires, versus using an `[or]`?

Reply:
> That's what I would have thought, but not necessarily. The [Paketo Node-Engine CNB](https://github.com/paketo-buildpacks/node-engine/blob/master/node/detect.go) does it this way. I find that clearer than with an `[or]`

That's very interesting. Given that we made mention to `[or]` somewhere above, I'd be interested in seeing a use case example where `[or]` _MUST_ be used for full clarity.

Reply:
I agree, and based on [this](https://buildpacks.slack.com/archives/CCP60GJLS/p1588625326288900?thread_ts=1588614869.288600&cid=CCP60GJLS) conversation, I'm not sure if the node-engine cnb should be doing that as well. I'm going to use the `jvm` cnb to illustrate using an `[or]` instead, as they do here: https://github.com/paketo-buildpacks/libjvm/blob/master/detect.go#L28-L44